modelmapper/stats.py

```diff
@@ -0,0 +1,474 @@
+        """
+        return False
+
+
+class TypeAccumulator(object):
+    """
+    Accumulates some value over every item it sees
+    """
+    def reset(self):
+        """
+        Resets the accumulator
+        """
+        pass
+
+    def inspect(self, field_name, item, settings):
+        """
+        Inspects the item
+        """
+        return False
+
+    def collect(self):
+        """
+        Sets the value of a field on the FieldStats object
+        """
+        pass
+
+
+class InSettingsMatcher(TypeMatcher):
+    """
+    Abstract Matacher -- This should be subclassed and not created directly.
+
+    Matches a value by comparing it to values in
+    the settings object.
+
+    >>> settings.test = ['1', 'y', 'true']
+    >>> InSettingsMatcher('test').match('test', settings)
+    False
+    >>> settings.test = ['1', 'y', 'true']
+    >>> InSettingsMatcher('test').match('1', settings)
+    True
+    """
+    def __init__(self, field):
+        self.field = field
+
+    def match(self, item, settings):
+        if not hasattr(settings, self.field):
+            # XXX: Raise exception
+            raise ValueError('missing field')
+            return False
+        return item in getattr(settings, self.field)
+
+
+class InMatcher(TypeMatcher):
+    """
+    Abstract Matacher -- This should be subclassed and not created directly.
+
+    Matches when a value contains a certain character
+
+    >>> InMatcher('$').match('test', settings)
+    False
+    >>> InMatcher('$').match('$0.0', settings)
+    True
+    """
+    def __init__(self, to_match):
+        self.to_match = to_match
+
+    def match(self, item, settings):
+        return self.to_match in item
+
+
+class NullMatcher(InSettingsMatcher):
+    """
+    Matches values to the designated null value strings in the settings file
+
+    >>> settings.null_values = ['null', '', ' ']
+    >>> NullMatcher().match('test', settings)
+    False
+    >>> settings.null_values = ['null', '', ' ']
+    >>> NullMatcher().match('null', settings)
+    True
+    """
+    value_type = HasNull
+
+    def __init__(self):
+        super().__init__('null_values')
+
+    def is_exclusive(self):
+        return True
+
+
+class BooleanMatcher(InSettingsMatcher):
+    """
+    Matches values to the designated boolean value strings in the settings file
+
+    >>> settings.booleans = ['1', 'y', 'true']
+    >>> BooleanMatcher().match('test', settings)
+    False
+    >>> settings.booleans = ['1', 'y', 'true']
+    >>> BooleanMatcher().match('true', settings)
+    True
+    """
+    value_type = HasBoolean
+
+    def __init__(self):
+        super().__init__('booleans')
+
+
+class DollarMatcher(InMatcher):
+    """
+    Matches values that contain the $ character
+
+    >>> DollarMatcher().match('test', settings)
+    False
+    >>> DollarMatcher().match('$1.50', settings)
+    True
+    """
+    value_type = HasDollar
+
+    def __init__(self):
+        super().__init__('$')
```

Comment:
If we end up with `__init__` taking more than one argument, running it with kwargs instead of args will be more readable.

Reply:
will update